### PR TITLE
Reimplement `--dump=raw-benchmark` as `-o raw-benchmark`

### PR DESCRIPTION
**Pull request by ajreynol**
Printing the original benchmark is simple, as it is exactly the commands we execute.

This removes the previous code from SmtEngine, which is currently broken.

### CODE REVIEW

File: src/smt/smt_engine.cpp [hunk 1]
```diff
@@ -680,8 +668,7 @@ void SmtEngine::defineFunctionsRec(
       lem = nm->mkNode(kind::FORALL, boundVars, lem, aexpr);
     }
     // assert the quantified formula
-    //   notice we don't call assertFormula directly, since this would
-    //   duplicate the output on raw-benchmark.
+    //   notice we don't call assertFormula directly.
```
Comment (nafur) [inline src/smt/smt_engine.cpp:671]: Is there another reason to not call `assertFormula` directly now?

**Review by nafur — COMMENTED** [review]
`--dump raw-benchmark` is also used in `datatype-dump.cvc` and `run_regression.py`

File: src/smt/smt_engine.cpp (hunk 1 above)
Comment (ajreynol) [reply src/smt/smt_engine.cpp:671]: Haha I thought you'd ask, the general reason is that we shouldn't use the SmtEngine interface from itself.

**Review by nafur — APPROVED** [review]
LGTM.

**Pull request merged**